Yafc.Model/Analysis/DependencyNode.cs

```diff
@@ -0,0 +1,296 @@
+
+        private AndNode(DependencyNode[] dependencies) => this.dependencies = dependencies; // Use Create
+
+        /// <summary>
+        /// Returns a <see cref="DependencyNode"/> that requires all of the children specified in <paramref name="dependencies"/>.
+        /// </summary>
+        /// <param name="dependencies">The children that should all be required by the returned value.</param>
+        internal static DependencyNode Create(IEnumerable<DependencyNode> dependencies) {
+            if (dependencies.Count() == 1) {
+                return dependencies.First();
+            }
+
+            List<DependencyNode> realDependencies = [];
+            foreach (DependencyNode item in dependencies) {
+                if (item is AndNode and) {
+                    realDependencies.AddRange(and.dependencies);
+                }
+                else {
+                    realDependencies.Add(item);
+                }
+            }
+            realDependencies = realDependencies.Distinct().ToList();
+            if (realDependencies.Count == 1) {
+                return realDependencies[0];
+            }
+            return new AndNode([.. realDependencies]);
+        }
+
+        internal override IEnumerable<FactorioId> Flatten() => dependencies.SelectMany(d => d.Flatten());
+
+        internal override bool IsAccessible(Func<FactorioId, bool> isAccessible) => dependencies.All(d => d.IsAccessible(isAccessible));
+        internal override Bits AggregateBits(Func<FactorioId, Bits> getBits) {
+            Bits result = default;
+            foreach (DependencyNode item in dependencies) {
+                result |= item.AggregateBits(getBits);
+            }
+            return result;
+        }
+        internal override AutomationStatus IsAutomatable(Func<FactorioId, AutomationStatus> isAutomatable, AutomationStatus automationState)
+            => dependencies.Min(d => d.IsAutomatable(isAutomatable, automationState));
+
+        public override void Draw(ImGui gui, Action<ImGui, DependencyList> builder) {
+            foreach (DependencyNode dependency in dependencies) {
+                dependency.Draw(gui, builder);
+            }
+        }
+    }
+
+    /// <summary>
+    /// A <see cref="DependencyNode"/> that requires at least one of its children.
+    /// </summary>
+    private sealed class OrNode : DependencyNode {
+        private readonly DependencyNode[] dependencies;
+
+        private OrNode(DependencyNode[] dependencies) => this.dependencies = dependencies; // Use Create
+
+        /// <summary>
+        /// Returns a <see cref="DependencyNode"/> that requires at least one of the children specified in <paramref name="dependencies"/>.
+        /// </summary>
+        /// <param name="dependencies">The children that will satisfy the returned value if at least one of them is satisfied.</param>
+        internal static DependencyNode Create(IEnumerable<DependencyNode> dependencies) {
```

Comment:
In this function there are the same considerations as for the previous 2 comments

Reply:
The unpacking code (calculating realDependencies, that converts from 1 to 2 below) would become more complicated without the checks that prevent nodes with only one child.
To a lesser extent, it's also easier to debug trees that have fewer nodes, though I didn't have to do much of that.

The special case for drawing a single-child `OrNode` can be removed, though.

![image](https://github.com/user-attachments/assets/88581f79-2dba-495b-9514-a803915c6413)

Reply:
> The unpacking code (calculating realDependencies, that converts from 1 to 2 below) would become more complicated without the checks that prevent nodes with only one child.

Makes sense, as it now can make some assumptions and be less complex. I guess we might need to write down these requirements somewhere, so we won't refactor this in the future to make things simpler (as I was thinking about)?